pkg/querier/chunks_handler.go

```diff
@@ -41,15 +41,13 @@ func ChunksHandler(queryable storage.Queryable) http.Handler {
 			return
 		}
 
-		//lazyQuerier, ok := querier.(lazyQuerier)
-		//if !ok {
-		//	http.Error(w, "not supported", http.StatusServiceUnavailable)
-		//	return
-		//}
-
-		//store, ok := lazyQuerier.next.(ChunkStore)
+		lazyQuerier, ok := querier.(lazyQuerier)
+		if !ok {
+			http.Error(w, "not supported", http.StatusServiceUnavailable)
+			return
+		}
 
-		store, ok := querier.(ChunkStore)
+		store, ok := lazyQuerier.next.(ChunkStore)
```

Comment:
this is pretty gross

Reply:
I've made `LazyQuerier` implement `ChunkStore` now.